### DISCUSSION TITLE
Add Group ID / GSO ID/ WSO ID

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#662] [Ideas]
I would like to request a new entry box to the 12 Step Meeting List plugin where we can store the ID of our groups. I am a webcoordinator for a fairly large area and a lot of my meetings have similar names. Having a box to enter the registration ID into AND having this box in the list overview, would enable me to search by this ID instead of the name.

I would appreciate it if this new box is also included in the import form, because I use this quite a bit when there are multiple changes to be uploaded.

If you have any questions about this, feel free to contact me.

Thanks for your hard work!

Nicky

**Comment by FlipperPA** [comment]
I would love to see both `gso_number` and `district_number`.

**Comment by OSAwebsite** [comment]
I believe we already have that, going by the filter in the top of the regular plugin list. It is where the 'Everywhere' column is, when you open the list attached to that box you can choose to toggle to the district. It is not, however, as a separate value present on the back end list. Which is why I have decided early on when I started to use the plugin, to change the group names. They show both their online status (I use the prefix '(elec') and their district (I use the suffix (D3) ). That has worked well for me in particular for the back end. But in general I don't expect to much in terms of data management in the actual backend of Wordpress. I am much quicker re-doing the whole list via an import. I am an Al-Anon, so I don't have to worry at this time about that messing with the companion app, which we have no access to. I can refresh my entire list of will into 300 meetings within 10 minutes and that is simply impossible via the Wordpress backend.

Having said that, having the group ID and the District ID as separate values visible in the back end list would help me to speed up those entries that I do correct/adjust manually...

**Comment by 97eighty** [comment]
@tech2serve @joshreisner Should this be discussed and hashed out in the [spec ](https://github.com/code4recovery/spec) first?

**Comment by joshreisner** [comment]
so @OSAwebsite is asking for a different way of enforcing uniqueness among groups, rather than the existing system of requiring unique `group` names. I would be careful conflating this with GSO's internal numbering system. Potential issues:

* GSO has different numbers for FNV and Netsuite
* GSO is only AA of course
* Local groups, eg NY Intergroup, often have different internal database numbers that they want to use, not the GSO ID#
* Local groups often have bizarre groupings, such as an intergroup having multiple subgroups within a group

Potential workaround: append the group number the group name, to make it unique. example: https://www.nyintergroup.org/meetings/midnite-15/ The group is called "Midnite" and they appended "#12920" which is their internal database number to make it unique across the tri-state area.

**Reply by tech2serve** [reply]
I suppose an option would be to create a generic identifier field in TSML, for which admin’s would also create the label for, that can be used in generally any manner the webmaster/meeting coordinator wishes.

We’ve had others ask to be able to enter GSO numbers….

If the team decided this was a go, I believe it should only be in TSML on the admin side, and would not go into the feed; therefore, it would not be shareable between sites. I’m also not clear on the coding we’d have to do to make it searchable as @OSAwebsite requested. 

Note: I converted this Issue to a Discussion based on our agreed workflow *before* I caught the date of the original request. My apologies if this causes confusion.

**Reply by Grateful-Viv** [reply]
I like the idea of having an additional field to capture and display a generic identifier. I’d use it to capture GSO Group numbers. But others can use it to record whatever they want, or nothing at all.

**Comment by 97eighty** [comment]
Ah, I misread the intent.

**Comment by OSAwebsite** [comment]
Hi team,

In my recent refresh of the website with my group and meeting data I have solved this problem.
I have moved the Id to the field that shows up underneath the checkboxes, in the edit mode of the meeting pages.
The biggest advantage being twofold:
That particular notes fields is included in a general search in the front end AND it is part of the email information when a person uses the 'request a change to this listing' feature. So my needs are met.

I decided to do this when I read that the GSO ID has a different status than the Al-Anon WSO ID. 
Many names in Al-Anon are identical or very similar so the group name has never been a good uniqueness identifier for me.
But the WSO ID does do that. It might be different in Europe or other parts of the world, but over here all groups are registered via WSO.

In the backend I use the unique post ID a lot. There is a plugin that reveals that number. And in my off-website files I can re-assemble the URL of each of my meetings using that posting number to directly jump to a particular meeting, front or back-end. Which means I hardly ever have do it through the Wordpress user interface. My group meetings, if they need editing, show up within seconds, whereas working within the Wordpress UI it can take minutes, depending on... I haven't got the foggiest.

I do know that the direct links are quick and accurate. The only downside is that when I do a complete refresh of the database (which happens maybe once or twice in a year), those numbers changed. But I have found a way to automate getting the new numbers into my database so across the board, I am a contented camper. Sometimes even a happy camper. I love it when technology works for me instead of me having to work due to technology....

Thanks for considering it, plugin team.

**Comment by OSAwebsite** [comment]
As a by the way and unconnected: are we OK to update Wordpress to 5.9?

**Reply by kiyote33** [reply]
I would hold up on moving to 5.9 until you have successfully validated your site still works correctly in a staging environment.  I've been testing several sites on my development machine which are all broken by WordPress 5.9.  The issues are mostly theme related, but I am also seeing some issues coming up with the 12 Step Meeting List plugin.

**Reply by Grateful-Viv** [reply]
I’ve upgraded to WP 5.9 with no issues.

**Reply by kiyote33** [reply]
Seems like a reinstall of 5.9 resolved my problems. Tested against 16 various themes today with no issues.

**Comment by OSAwebsite** [comment]
Ok thanks for the heads-up! I have a simple theme but will wait for a bit.

On Tue, Feb 1, 2022 at 12:49 PM kiyote33 ***@***.***> wrote:

> I would hold up on moving to 5.9 until you have successfully validated
> your site still works correctly in a staging environment. I've been testing
> several sites on my development machine which are all broken by WordPress
> 5.9. The issues are mostly theme related, but I am also seeing some issues
> coming up with the 12 Step Meeting List plugin.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/662#discussioncomment-2090877>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOIWBF75OHTJXFGHXK55Y4DUZAMKBANCNFSM5NGYH6RA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/662/comments/2090877@
> github.com>
>
-- 


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
As long as it is included when doing a free search AND in the import sheet.
That would really help me.

On Tue, Feb 1, 2022 at 5:25 PM Grateful-Viv ***@***.***>
wrote:

> I like the idea of having an additional field to capture and display a
> generic identifier. I’d use it to capture GSO Group numbers. But others can
> use it to record whatever they want, or nothing at all.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4recovery/12-step-meeting-list/discussions/662#discussioncomment-2092313>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOIWBF35ZIGD5OW2TV2X7SDUZBMUNANCNFSM5NGYH6RA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> <code4recovery/12-step-meeting-list/repo-discussions/662/comments/2092313@
> github.com>
>
-- 


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Comment by OSAwebsite** [comment]
OK, I will do the staging site first and when there are no issues do the
live site as well.


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
